packages/merkle_tree/src/tests/merkle_proof/test_with_pedersen.cairo

```diff
@@ -0,0 +1,219 @@
+        0x05fb6a626bb2c1e12fc2d6fa7f218ec06928ba5febf4d5677c2c5060827e383b
+    ].span();
+
+    // `proof_flags.len()` is not equal to `proof.len() + leaves_to_prove.len() + 1`
+    let proof_flags = [true, false].span();
+
+    verify_multi_proof::<PedersenCHasher>(proof, proof_flags, root, leaves_to_prove);
+}
+
+#[test]
+#[should_panic(expected: ('Index out of bounds',))]
+fn test_process_multi_proof_flags_extra_leaves_expected() {
+    let leaves = LEAVES();
+    let leaves_to_prove = [
+        leaf_hash(*leaves.at(0)), leaf_hash(*leaves.at(1)), leaf_hash(*leaves.at(2))
+    ].span();
+    let proof = [
+        0x044fdc540a81d0189ed30b49d64136f9e8bd499c942ba170404ef0b9406e524c,
+        0x05fb6a626bb2c1e12fc2d6fa7f218ec06928ba5febf4d5677c2c5060827e383b
+    ].span();
+
+    // For each true one leaf is expected
+    let proof_flags = [true, true, true, true].span();
+
+    process_multi_proof::<PedersenCHasher>(proof, proof_flags, leaves_to_prove);
+}
+
+#[test]
+#[should_panic(expected: ('Index out of bounds',))]
+fn test_process_multi_proof_flags_extra_proofs_expected() {
+    let leaves = LEAVES();
+    let leaves_to_prove = [
+        leaf_hash(*leaves.at(0)), leaf_hash(*leaves.at(1)), leaf_hash(*leaves.at(2))
+    ].span();
+    let proof = [
+        0x044fdc540a81d0189ed30b49d64136f9e8bd499c942ba170404ef0b9406e524c,
+        0x05fb6a626bb2c1e12fc2d6fa7f218ec06928ba5febf4d5677c2c5060827e383b
+    ].span();
+
+    // For each false one proof is expected
+    let proof_flags = [true, false, false, false].span();
+
+    process_multi_proof::<PedersenCHasher>(proof, proof_flags, leaves_to_prove);
+}
+
+#[test]
+#[should_panic(expected: ('Index out of bounds',))]
+fn test_verify_multi_proof_flags_extra_leaves_expected() {
+    let leaves = LEAVES();
+    let leaves_to_prove = [
+        leaf_hash(*leaves.at(0)), leaf_hash(*leaves.at(1)), leaf_hash(*leaves.at(2))
+    ].span();
+    let root = 0x02a40717603180fa52f40a55508cd360d301840f3e502665cf0132ef412911de;
+    let proof = [
+        0x044fdc540a81d0189ed30b49d64136f9e8bd499c942ba170404ef0b9406e524c,
+        0x05fb6a626bb2c1e12fc2d6fa7f218ec06928ba5febf4d5677c2c5060827e383b
+    ].span();
+
+    // For each true one leaf is expected
+    let proof_flags = [true, true, true, true].span();
+
+    verify_multi_proof::<PedersenCHasher>(proof, proof_flags, root, leaves_to_prove);
+}
+
+#[test]
+#[should_panic(expected: ('Index out of bounds',))]
+fn test_verify_multi_proof_flags_extra_proofs_expected() {
+    let leaves = LEAVES();
+    let leaves_to_prove = [
+        leaf_hash(*leaves.at(0)), leaf_hash(*leaves.at(1)), leaf_hash(*leaves.at(2))
+    ].span();
+    let root = 0x02a40717603180fa52f40a55508cd360d301840f3e502665cf0132ef412911de;
+    let proof = [
+        0x044fdc540a81d0189ed30b49d64136f9e8bd499c942ba170404ef0b9406e524c,
+        0x05fb6a626bb2c1e12fc2d6fa7f218ec06928ba5febf4d5677c2c5060827e383b
+    ].span();
```

Comment:
The same root and proof values are used in almost all the test cases. Should we consider them constants and place them in the `common` module alongside with `LEAVES`?

Reply:
I agree, but `common` module is for things that are common among different modules. This can be defined as local test constants.